File: internal/adapters/http/dto/category.go
package dto

type CreateCategoryRequest struct {
	Name             string `json:"name"`
	Direction        string `json:"direction"` // IN or OUT
	IsBudgetRelevant bool   `json:"is_budget_relevant"`
}

type CategoryResponse struct {
	ID               int32  `json:"id"`
	Name             string `json:"name"`
	Direction        string `json:"direction"`
	IsBudgetRelevant bool   `json:"is_budget_relevant"`
	IsActive         bool   `json:"is_active"`
}

File: frontend/layers/categories/types/category.ts
export type Direction = 'IN' | 'OUT'

export interface Category {
  id: number
  name: string
  direction: Direction
  is_budget_relevant: boolean
  is_active: boolean
  inactive_from_month?: string
}

export interface CreateCategoryRequest {
  name: string
  direction: Direction
  is_budget_relevant: boolean
}

export interface UpdateCategoryRequest {
  name: string
  direction: Direction
  is_budget_relevant: boolean
  is_active: boolean
}
